packages/nimble-components/src/select/template.ts

```diff
@@ -81,21 +87,49 @@ SelectOptions
             horizontal-positioning-mode="locktodefault"
             horizontal-scaling="anchor"
             ?hidden="${x => (x.collapsible ? !x.open : false)}">
-            <div
-                class="listbox"
-                id="${x => x.listboxId}"
-                part="listbox"
-                role="listbox"
-                ?disabled="${x => x.disabled}"
-                ${ref('listbox')}
-            >
-                <slot
-                    ${slotted({
+            <div class="listbox-background">
+                <div
+                    class="
+                        listbox 
+                        ${x => (x.filteredOptions.length === 0 ? 'empty' : '')}
+                        ${x => x.positionAttribute}
+                    "
+                    id="${x => x.listboxId}"
+                    part="listbox"
+                    role="listbox"
+                    ?disabled="${x => x.disabled}"
+                    ${ref('listbox')}
+                >
+                    ${when(x => x.filterMode !== FilterMode.none, html<Select>`
+                        <div class="search-field ${x => x.positionAttribute}">
+                            <${iconMagnifyingGlassTag} class="search-icon"></${iconMagnifyingGlassTag}>
+                            <input
+                                class="filter-input"
+                                ?disabled="${x => x.disabled}"
+                                @input="${(x, c) => x.inputHandler(c.event as InputEvent)}"
+                                @click="${(x, c) => x.inputClickHandler(c.event as MouseEvent)}"
+                                ${ref('searchInput')}
+                                placeholder="${x => filterSearchLabel.getValueFor(x)}"
+                                value="${x => x.filter}"
+                            />
+                        </div>
+                    `)}
+                    <div ${ref('scrollableElement')}
+                        class="scrollable-element">
+                        <slot
+                            ${slotted({
         filter: (n: Node) => n instanceof HTMLElement && Listbox.slottedOptionFilter(n),
         flatten: true,
         property: 'slottedOptions',
     })}
-                ></slot>
+                        ></slot>
+                    </div>
+                    ${when(x => (x.filterMode !== FilterMode.none && x.filteredOptions.length === 0), html<Select>`
+                        <span class="no-results-label ${x => x.positionAttribute}">
```

Comment:
The alignment of this text seems off to me. It appears to be aligned with the left edge of a list-option rather than with the left edge of the content within a list-option.